root/static/scripts/edit/URLCleanup.js

```diff
@@ -1102,16 +1102,54 @@ const CLEANUPS = {
     },
   },
   'classicalarchives': {
-    match: [new RegExp(
-      '^(https?://)?(www\\.)?classicalarchives\\.com/' +
-      '(album|artist|composer|ensemble|work)/',
-      'i',
-    )],
+    match: [
+      new RegExp(
+        '^(https?://)?(www\\.)?classicalarchives\\.com/' +
+        '(album|artist|composer|ensemble|work)/',
+        'i',
+      ),
+      new RegExp(
+        '^(https?://)?(www\\.)?classicalarchives\\.com/newca/#!/' +
+        '(Composer|Performer|Work)/',
+        'i',
+      ),
+    ],
     type: LINK_TYPES.otherdatabases,
     clean: function (url) {
-      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?classicalarchives\.com\/(album|artist|composer|ensemble|work)\/([^\/?#]+)(?:.*)?$/, 'https://www.classicalarchives.com/$1/$2');
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?classicalarchives\.com\//, 'https://www.classicalarchives.com/');
+      /*
+       * Both newca and "old-style" links work, old redirects to new
+       * unless requested otherwise. CA claimed they'll support both
+       * going forward, so mapping all to old-style for now.
+       * newca entities match except Performer, where the old type
+       * is determined by the first letter of the ID: ensemble (e)
+       * or artist (p)
+       * newca Album links are blocked for now because they can't be converted
+       */
+      url = url.replace(/^https:\/\/www\.classicalarchives\.com\/newca\/#!\/Composer\/([^\/?#]+)/, 'https://www.classicalarchives.com/composer/$1.html');
+      url = url.replace(/^https:\/\/www\.classicalarchives\.com\/newca\/#!\/Work\/([^\/?#]+)/, 'https://www.classicalarchives.com/work/$1.html');
+      url = url.replace(/^https:\/\/www\.classicalarchives\.com\/newca\/#!\/Performer\/e([^\/?#]+)/, 'https://www.classicalarchives.com/ensemble/$1.html');
+      url = url.replace(/^https:\/\/www\.classicalarchives\.com\/newca\/#!\/Performer\/p([^\/?#]+)/, 'https://www.classicalarchives.com/artist/$1.html');
+      url = url.replace(/^https:\/\/www\.classicalarchives\.com\/(album|artist|composer|ensemble|work)\/([^\/?#]+)(?:.*)?$/, 'https://www.classicalarchives.com/$1/$2');
```

Comment:
Since you normalized URL’s scheme and authority above, and no more change is being made to them, you can just capture them instead of rewriting them, so as to make actual changes more visible:

```suggestion
      url = url.replace(/^(https:\/\/www\.classicalarchives\.com)\/newca\/#!\/Composer\/([^\/?#]+)/, '$1/composer/$2.html');
      url = url.replace(/^(https:\/\/www\.classicalarchives\.com)\/newca\/#!\/Work\/([^\/?#]+)/, '$1/work/$2.html');
      url = url.replace(/^(https:\/\/www\.classicalarchives\.com)\/newca\/#!\/Performer\/e([^\/?#]+)/, '$1/ensemble/$2.html');
      url = url.replace(/^(https:\/\/www\.classicalarchives\.com)\/newca\/#!\/Performer\/p([^\/?#]+)/, '$1/artist/$2.html');
      url = url.replace(/^(https:\/\/www\.classicalarchives\.com)\/(album|artist|composer|ensemble|work)\/([^\/?#]+)(?:.*)?$/, '$1/$2/$3');
```

Reply:
Good idea!